src/spdx_tools/common/typing/dataclass_with_properties.py

```diff
@@ -26,11 +28,11 @@ def set_field(self, value: field_type):
     def set_field_with_better_error_message(self, value: field_type):
         try:
             set_field(self, value)
-        except TypeError as err:
-            error_message: str = f"SetterError {self.__class__.__name__}: {err.args[0]}"
+        except TypeCheckError as err:
+            error_message: str = f"SetterError {self.__class__.__name__}: {err}"
             # As setters are created dynamically, their argument name is always "value". We replace it by the
             # actual name so the error message is more helpful.
-            raise TypeError(error_message.replace("value", field_name, 1) + f": {value}")
```

Comment:
Is there a particular reason why we don't print out the actual value anymore? I think this is quite useful to quickly pinpoint the source of the error.

Reply:
There are two main reasons for this change, but they might be a matter of taste:
- The "new" messages contain the type of the falsy value, so I thought of it as a duplicate, e.g. 
<pre>'SetterError Package: argument "name" (None) is not an instance of str'</pre>
vs
<pre>'SetterError Package: argument "name" (None) is not an instance of str: None'</pre>
- If the type hint contains a `Union` the error message is quite long and appending the real value doesn't really improve readability in my opinion, 
e.g.
<pre>
SetterError Package: argument "
            '"download_location" (int) did not match any element in the union:\\n  str: '
            "is not an instance of str\\n  "
            "spdx_tools.spdx.model.spdx_no_assertion.SpdxNoAssertion: is not an instance "
            "of spdx_tools.spdx.model.spdx_no_assertion.SpdxNoAssertion\\n  "
            "spdx_tools.spdx.model.spdx_none.SpdxNone: is not an instance of "
            "spdx_tools.spdx.model.spdx_none.SpdxNone: 5'
</pre>

Reply:
I implemented the string manipulation and added the value back in again.